src/main/java/stringCalculator/Operation.java

```diff
@@ -0,0 +1,36 @@
+package stringCalculator;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+public enum Operation {
+    ADDING("+", (number1, number2) -> number1 + number2),
+    SUBTRACTION("-", (number1, number2) -> number1 - number2),
+    MULTIPLICATION("*", (number1, number2) -> number1 * number2),
+    DIVISION("/", (number1, number2) -> number1 / number2);
+
+    private String opterator;
+
+    private BiFunction<Integer, Integer, Integer> fomular;
+
+    Operation(String opterator, BiFunction<Integer, Integer, Integer> fomular) {
+        this.opterator = opterator;
+        this.fomular = fomular;
+    }
+
+
+    public static Operation getOperation(String opterator) {
+        Optional<Operation> operation = Arrays.stream(Operation.values()).filter(obj -> obj.opterator.equals(opterator)).findAny();
+        if (!operation.isPresent()) {
+            throw new IllegalArgumentException("calculating param is wrong : contained not number or wrong opterator");
+        }
+        return operation.get();
```

Comment:
Optional을 생략하고 다음과 같이도 변경해볼 수 있을 것 같아요 :)
```suggestion
         Arrays.stream(Operation.values())
                    .filter(obj -> obj.opterator.equals(opterator))
                    .findFirst()
                    .orElseThrow(() -> 
                          new IllegalArgumentException("calculating param is wrong : contained not number or wrong opterator"));
```

Reply:
와.. 저 처음에 수정할때 findAny로 해서 elseThrow 보고 IllegalArgumentException 던지는 부분까진했었는데 메시지를 람다로 넘기면되는거였군요!.. 좀더 읽어봤어야했는데.. 후딱하고 퇴근하려다보니 옵셔널로 해버렸습니다 ^^;